src/transformers/generation/utils.py

```diff
@@ -1452,10 +1452,11 @@ def _prepare_generated_length(
         ):
             generation_config.max_length -= inputs_tensor.shape[1]
         elif has_default_max_length:  # by default let's always generate 20 new tokens
-            generation_config.max_length = generation_config.max_length + input_ids_length
-            max_position_embeddings = getattr(self.config, "max_position_embeddings", None)
-            if max_position_embeddings is not None:
-                generation_config.max_length = min(generation_config.max_length, max_position_embeddings)
+            if generation_config.max_length == GenerationConfig().max_length:
```

Comment:
if `generation_config.max_length` is not the same as the default value (`20`), let's not change it!!!!

Reply:
should we not change `has_default_max_length` directly? 🤗

Reply:
I would rather avoid changing its original definition in this PR as it is also used in several places.